### PR TITLE
Adding CSV parsing support for FX Forwards

### PR DESCRIPTION
**Pull request by andreiruse**

### CODE REVIEW

**Comment by marc-henrard** [comment]
I have added a couple of tests for different combinations of pay/receive. Look good to me.

File: modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+   * Parses the data from a CSV row.
+   *
+   * @param row  the CSV row object
+   * @param info  the trade info object
+   * @param resolver  the resolver used to parse additional information. This is not currently used in this method.
+   * @return the parsed trade, as an instance of {@link FxSingleTrade}
+   */
+  private static FxSingleTrade parseRow(CsvRow row, TradeInfo info, TradeCsvInfoResolver resolver) {
+    LocalDate paymentDate = LoaderUtils.parseDate(row.getField(PAYMENT_DATE_HEADER));
+    PayReceive leg1Direction = PayReceive.of(row.getField(LEG_1_DIRECTION_HEADER));
```
Comment (jodastephen) [inline modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java]: `LoaderUtils.parsePayReceive()`

File: modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    Currency leg1Currency = Currency.of(row.getField(LEG_1_CURRENCY_HEADER));
+    double leg1Notional = Double.parseDouble(row.getField(LEG_1_NOTIONAL_HEADER));
```
Comment (jodastephen) [inline modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java]: `LoaderUtils.parseDouble()`

File: modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    PayReceive leg2Direction = PayReceive.of(row.getField(LEG_2_DIRECTION_HEADER));
+    Currency leg2Currency = Currency.of(row.getField(LEG_2_CURRENCY_HEADER));
+    double leg2Notional = Double.parseDouble(row.getField(LEG_2_NOTIONAL_HEADER));
+
```
Comment (jodastephen) [inline modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java]: Validate that we don't have two pay legs or two receive legs

File: modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  /**
+   * Parses the data from a CSV row.
+   *
+   * @param row  the CSV row object
+   * @param info  the trade info object
+   * @param resolver  the resolver used to parse additional information. This is not currently used in this method.
+   * @return the parsed trade, as an instance of {@link FxSingleTrade}
+   */
+  private static FxSingleTrade parseRow(CsvRow row, TradeInfo info, TradeCsvInfoResolver resolver) {
+    LocalDate paymentDate = LoaderUtils.parseDate(row.getField(PAYMENT_DATE_HEADER));
```
Comment (jodastephen) [inline modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java]: Should also support business day adjustment fields as per a swap.

Comment (andreiruse) [reply modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java]: @marc-henrard How do you suggest we add this to the CSV format (header)?

Comment (jodastephen) [reply modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java]: See `FullSwapCsvLoader`.

```
  private static final String PAYMENT_DATE = "Payment Date";
  private static final String PAYMENT_DATE_CNV_FIELD = "Payment Date Convention";
  private static final String PAYMENT_DATE_CAL_FIELD = "Payment Date Calendar";
```

**Pull request merged**